pkg/gadgets/trace/signal/tracer/tracer.go

```diff
@@ -24,17 +24,23 @@ import (
 	"github.com/cilium/ebpf/perf"
 	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
 	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
-	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/sigsnoop/tracer"
-	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/sigsnoop/types"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/signal/types"
 	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
 
 	"golang.org/x/sys/unix"
 )
 
 //go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang sigsnoop ./bpf/sigsnoop.bpf.c -- -I../../../../${TARGET}
 
+type Config struct {
+	MountnsMap   *ebpf.Map
```

Comment:
I think it's possible but IMO the advantage is too low, actually it'll complicate the callers as they would have to create the `GenericConfig` to embed it within each gadget `Config`.

Reply:
I agree with Mauricio, I think embedding is useful to avoid having `commonFlags.OutputConfig.Verbose` as you suggested [here](https://github.com/kinvolk/inspektor-gadget/blob/04c2a3664d830cfc39c294f9a37b1b73159de008/cmd/kubectl-gadget/utils/flags.go#L44-L45). Or when we want to promote the embedded struct's methods to the embedding struct, like what many gadgets' parser do with the [`BaseParser`](https://github.com/kinvolk/inspektor-gadget/blob/f61e3d56205205df454b22562967d30f7fe6c8ad/cmd/common/snapshot/process.go#L36).